src/OpenProject.Shared/Math3D/Camera.cs

```diff
@@ -0,0 +1,31 @@
+﻿using OpenProject.Shared.Math3D.Enumeration;
+
+namespace OpenProject.Shared.Math3D
+{
+  /// <summary>
+  /// A camera wrapper using decimal precision.
+  /// </summary>
+  public abstract class Camera
+  {
+    /// <summary>
+    /// The camera location, also named camera viewpoint.
+    /// </summary>
+    public Vector3 Viewpoint { get; set; } = new Vector3(0, 0, 0);
+
+    /// <summary>
+    /// The camera view direction.
+    /// </summary>
+    public Vector3 Direction { get; set; } = new Vector3(0, 0, 0);
+
+    /// <summary>
+    /// The camera up vector.
+    /// </summary>
+    public Vector3 UpVector { get; set; } = new Vector3(0, 0, 0);
+
+    /// <summary>
+    /// The camera type, which can be orthogonal or perspective.
+    /// No camera type (None) is considered invalid camera data.
+    /// </summary>
+    public CameraType Type { get; set; } = CameraType.None;
```

Comment:
Should it really initialize to an invalid state? Maybe we could add a constructor that take the `CameraType` enum, so that callers are forced to decide and can't forget it.

Reply:
Or, better yet, since I just saw the inherited classes below, maybe it a `public abstract CameraType Type { get; }` and have the concrete implementations return the correct enum.

Reply:
here is the thing :D The camera doesn't really fit my definition of an immutable. Hence, I'm always a fan, of having only dummy constructors for classes, that are completely mutable. But you made a fine point. I first added the enumeration and later introduced the inherited classes. I like you proposal, so let's see, how we can make it more easy to understand and use.